File: backend/app/schemas/template.py
from pydantic import BaseModel
from typing import Optional


class TemplateCreate(BaseModel):
    name: str
    description: str = ""
    tool: str
    oracle_version: str = "19c"
    params: dict = {}


class TemplateUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    tool: Optional[str] = None
    oracle_version: Optional[str] = None
    params: Optional[dict] = None


class TemplateResponse(BaseModel):
    id: int
    name: str
    description: str
    tool: str
    oracle_version: str
    params: dict
    is_builtin: bool
    created_at: str
    updated_at: str

File: frontend/src/api/template.ts
import api from './index'

export interface TemplateItem {
  id: number
  name: string
  description: string
  tool: string
  oracle_version: string
  params: Record<string, any>
  is_builtin: boolean
  created_at: string
  updated_at: string
}

export function getTemplates(tool?: string) {
  return api.get('/templates', { params: { tool } })
}

export function getTemplate(id: number) {
  return api.get(`/templates/${id}`)
}

export function createTemplate(data: { name: string; description?: string; tool: string; oracle_version?: string; params: any }) {
  return api.post('/templates', data)
}

export function updateTemplate(id: number, data: any) {
  return api.put(`/templates/${id}`, data)
}

export function deleteTemplate(id: number) {
  return api.delete(`/templates/${id}`)
}

export function seedBuiltinTemplates() {
  return api.post('/templates/seed')
}
